modules/menu.rb

```diff
@@ -26,5 +26,19 @@ def menu
     retry if (retries += 1) < 3
     'Unable to fetch menu'
   end
+
+  def restaurant_menu
+    retries ||= 0
```

Comment:
Their is a fair bit of code duplication going on here. Could this possibly be extracted to a seperate function to witch you perhaps could provide a &block where function specific actions are executed?